full-backend-tests/src/test/java/org/graylog/plugins/views/SuggestionResourceIT.java

```diff
@@ -38,46 +40,39 @@
-        this.stream2Id = StreamUtils.createStream(requestSpec, "Stream #2", defaultIndexSetId, new StreamUtils.StreamRule(StreamRuleType.EXACT.toInteger(), "stream2", "target_stream", false));
-
-        GelfInputUtils.createGelfHttpInput(mappedPort, GELF_HTTP_PORT, requestSpec);
-        GelfInputUtils.postMessage(mappedPort,
-                "{\"short_message\":\"SuggestionResourceIT#1\", \"host\":\"example.org\", \"facility\":\"junit\", \"_target_stream\": \"stream1\"}",
-                requestSpec);
-        GelfInputUtils.postMessage(mappedPort,
-                "{\"short_message\":\"SuggestionResourceIT#2\", \"host\":\"example.org\", \"facility\":\"test\", \"_target_stream\": \"stream1\"}",
-                requestSpec);
-        GelfInputUtils.postMessage(mappedPort,
-                "{\"short_message\":\"SuggestionResourceIT#3\", \"host\":\"example.org\", \"facility\":\"test\", \"_target_stream\": \"stream1\"}",
-                requestSpec);
-        GelfInputUtils.postMessage(mappedPort,
-                "{\"short_message\":\"SuggestionResourceIT#4\", \"host\":\"foreign.org\", \"facility\":\"test\", \"_target_stream\": \"stream2\"}",
-                requestSpec);
-         SearchUtils.waitForMessage(requestSpec, "SuggestionResourceIT#1");
-         SearchUtils.waitForMessage(requestSpec, "SuggestionResourceIT#2");
-         SearchUtils.waitForMessage(requestSpec, "SuggestionResourceIT#3");
-         SearchUtils.waitForMessage(requestSpec, "SuggestionResourceIT#4");
+
+        final String defaultIndexSetId = api.indices().defaultIndexSetId();
+        this.stream1Id = api.streams().createStream("Stream #1", defaultIndexSetId, new Streams.StreamRule(StreamRuleType.EXACT.toInteger(), "stream1", "target_stream", false));
+        this.stream2Id = api.streams().createStream("Stream #2", defaultIndexSetId, new Streams.StreamRule(StreamRuleType.EXACT.toInteger(), "stream2", "target_stream", false));
+
+        api.gelf().createGelfHttpInput()
+                .postMessage(
+                        "{\"short_message\":\"SuggestionResourceIT#1\", \"host\":\"example.org\", \"facility\":\"junit\", \"_target_stream\": \"stream1\"}")
+                .postMessage(
+                        "{\"short_message\":\"SuggestionResourceIT#2\", \"host\":\"example.org\", \"facility\":\"test\", \"_target_stream\": \"stream1\"}")
+                .postMessage(
+                        "{\"short_message\":\"SuggestionResourceIT#3\", \"host\":\"example.org\", \"facility\":\"test\", \"_target_stream\": \"stream1\"}")
+                .postMessage("{\"short_message\":\"SuggestionResourceIT#4\", \"host\":\"foreign.org\", \"facility\":\"test\", \"_target_stream\": \"stream2\"}");
+
+        api.search().waitForMessage("SuggestionResourceIT#1");
+        api.search().waitForMessage("SuggestionResourceIT#2");
+        api.search().waitForMessage("SuggestionResourceIT#3");
+        api.search().waitForMessage("SuggestionResourceIT#4");
```

Comment:
It could be chained if you agree with my comment from Search class.

Reply:
Took it one step further and allowed waiting for multiple messages in one go. Should save some network roundtrips in optimal case.